### PR TITLE
[configdb] Migrate minigraph configurations to DB

### PR DESCRIPTION
**Pull request by taoyl-ms**
- Modify minigraph parser output format so it fit DB schema
- Modify configuration templates to fit new schema
- Systemd services dependencies are modified so database starts before any configuration consumer

### CODE REVIEW

**Comment by lguohan** [comment]
build failure.

**Comment by taoyl-ms** [comment]
This is depending on [https://github.com/Azure/sonic-py-swsssdk/pull/12](https://github.com/Azure/sonic-py-swsssdk/pull/12). I'll need to merge that PR first and update submodule pointer to make this build.

---
In reply to: [328136365](https://github.com/Azure/sonic-buildimage/pull/942#issuecomment-328136365) [](ancestors = 328136365)

File: src/sonic-config-engine/tests/test_cfggen.py [hunk 1]
```diff
@@ -76,47 +76,47 @@ def test_minigraph_acl(self):
...
+        self.assertEqual(output.strip(), "[('PortChannel01', 'FC00::71/126'), ('PortChannel01', '10.0.0.56/31')]")
 
     def test_minigraph_neighbors(self):
-        argument = '-m "' + self.sample_graph_t0 + '" -p "' + self.port_config + '" -v minigraph_neighbors'
+        argument = '-m "' + self.sample_graph_t0 + '" -p "' + self.port_config + '" -v "DEVICE_NEIGHBOR[\'ARISTA01T1\']"'
         output = self.run_script(argument)
-        self.assertEqual(output.strip(), "{'Ethernet116': {'name': 'ARISTA02T1', 'port': 'Ethernet1/1'}, 'Ethernet124': {'name': 'ARISTA04T1', 'port': 'Ethernet1/1'}, 'Ethernet112': {'name': 'ARISTA01T1', 'port': 'Ethernet1/1'}, 'Ethernet120': {'name': 'ARISTA03T1', 'port': 'Ethernet1/1'}}")
+        self.assertEqual(output.strip(), "{'mgmt_addr': None, 'hwsku': 'Arista', 'lo_addr': None, 'local_port': 'Ethernet112', 'type': 'LeafRouter', 'port': 'Ethernet1/1'}")
 
     def test_minigraph_peers_with_range(self):
```
Comment (lguohan) [inline src/sonic-config-engine/tests/test_cfggen.py]: >test_minigraph_peers_with_range [](start = 8, length = 31)

I think we need to test every key/value in the minigraph here.
For example, the SYSLOG_SERVERS, NTP_SERVER,

File: src/sonic-config-engine/minigraph.py [hunk 2]
```diff
@@ -458,90 +385,75 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
-    if devices != None:
-        results['minigraph_console'] = get_console_info(devices, console_dev, console_port)
-        results['minigraph_mgmt'] = get_mgmt_info(devices, mgmt_dev, mgmt_port)
-    results['minigraph_hostname'] = hostname
-    results['inventory_hostname'] = hostname
-    results['syslog_servers'] = syslog_servers
-    results['dhcp_servers'] = dhcp_servers
-    results['ntp_servers'] = ntp_servers
-    results['forced_mgmt_routes'] = mgmt_routes
     results['erspan_dst'] = erspan_dst
```
Comment (lguohan) [inline src/sonic-config-engine/minigraph.py]: so, we are not converting this one?

File: src/sonic-config-engine/minigraph.py [hunk 3]
```diff
@@ -458,90 +385,75 @@ def parse_xml(filename, platform=None, port_config_file=None):
...
-    results['deployment_id'] = deployment_id
     results['ethernet_interfaces'] = ethernet_interfaces
```
Comment (lguohan) [inline src/sonic-config-engine/minigraph.py]: is this one in config db?

**Pull request merged**